--- /usr/share/voicecall-ui-jolla/pages/calllog/CallLogItem.qml 
+++ /usr/share/voicecall-ui-jolla/pages/calllog/CallLogItem.qml
@@ -4,6 +4,7 @@ import org.nemomobile.commhistory 1.0
 import org.nemomobile.contacts 1.0
 import Sailfish.Contacts 1.0
 import "../../common/CallLog.js" as CallLog
+import "../../common/phonelocation.js" as Location
 
 ListItem {
     id: root
@@ -79,13 +80,23 @@ ListItem {
                 width: Math.min(implicitWidth, parent.width)
             }
             Label {
+                id:secondNameText
                 text: people.populated && person ? person.secondaryName : ""
                 color: highlighted ? Theme.secondaryHighlightColor : secondaryColor
                 truncationMode: TruncationMode.Fade
                 width: Math.min(implicitWidth, parent.width - firstNameText.width)
                 visible: width > 0
             }
-        }
+    Label {
+        id:location
+        text: Location.getLocation(CallLog.formatNumber(remoteUid).toString())
+        color: highlighted ? Theme.secondaryHighlightColor : secondaryColor
+       font.pixelSize: Theme.fontSizeSmall*2/3
+        anchors.verticalCenter: parent.verticalCenter
+       width: Math.min(implicitWidth, parent.width - firstNameText.width -secondNameText.width)
+        visible: width > 0
+    }
+ }
         Label {
             id: timeStampLabel
             // JS comma operator used to ensure relative timestamp is updated whenever Qt.application.active changes
@@ -107,6 +118,20 @@ ListItem {
         color: highlighted ? Theme.secondaryHighlightColor : secondaryColor
     }
 
+    Label {//custom
+       id: phoneNumber
+        anchors {
+            top: description.bottom
+            left: description.left
+        }
+        font.pixelSize: Theme.fontSizeExtraSmall
+        text: people.populated || remoteUid !== undefined
+                          ? (!person || (person.primaryName.length === 0 && person.secondaryName.length === 0)
+                                 ? "" 
+                                : CallLog.formatNumber(remoteUid))
+                          : ""
+        color: highlighted ? Theme.secondaryHighlightColor : secondaryColor
+    }
     Image {
         id: icon
         anchors.top: description.top
--- /usr/share/voicecall-ui-jolla/pages/calling/IncomingCallViewBase.qml
+++ /usr/share/voicecall-ui-jolla/pages/calling/IncomingCallViewBase.qml
@@ -1,6 +1,7 @@
 import QtQuick 2.0
 import Sailfish.Silica 1.0
 import "../../common/CallLog.js" as CallLog
+import "../../common/phonelocation.js" as Location
 
 SilicaFlickable {
     id: incomingCallView
@@ -203,6 +204,14 @@ SilicaFlickable {
                 anchors.horizontalCenter: parent.horizontalCenter
                 source: "image://theme/icon-l-redirect"
             }
+            Label{
+                id:location
+                text:Location.getLocation(CallLog.formatNumber(phoneNumber).toString())
+                font { family: Theme.fontFamilyHeading; pixelSize: Theme.fontSizeLarge }
+                maximumLineCount: 2
+                wrapMode: Text.Wrap
+                anchors.horizontalCenter: parent.horizontalCenter
+            }
             Label {
                 width: parent.width
                 visible: text.length > 0
